src/Zap.sol

```diff
@@ -343,25 +369,38 @@ contract Zap is Errors {
     /// @custom:synthetix RBAC permission required: "PERPS_MODIFY_COLLATERAL"
     /// @param _accountId synthetix perp market account id
     /// @param _collateralId synthetix market id of collateral
+    /// @param _collateralAmount amount of collateral to unwind
     /// @param _zapTolerance acceptable slippage for zapping
+    /// @param _unwrapTolerance acceptable slippage for unwrapping
     /// @param _swapTolerance acceptable slippage for swapping
     /// @param _receiver address to receive unwound collateral
     function unwind(
         uint128 _accountId,
         uint128 _collateralId,
+        uint256 _collateralAmount,
         uint256 _zapTolerance,
+        uint256 _unwrapTolerance,
         uint256 _swapTolerance,
         address _receiver
     )
         external
+        isAuthorized(_accountId)
+        requireStage(Stage.UNSET)
```

Comment:
is it worth using the traditional `nonReentrant` modifier for external funcs in addition to the multi-stage guard, or is that overkill?

Reply:
i think it is overkill. tbh I think even the multi-stage thing is a bit overkill but I can live with it